BlueLevel/UI/target_ui.gd

```diff
@@ -0,0 +1,30 @@
+extends CanvasLayer
+
+var progress_bar : TextureProgressBar
+@export var time_left : float = 30.0
+@export var max_time : float = 30.0  # Set the maximum time
+var targetCount : int
+
+func _ready():
```

Comment:
Could add "BlueGlobals.score = 0" in the ready function to ensure the score is reset when the level loads.